packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt

```diff
@@ -32,174 +21,177 @@ import expo.modules.imagepicker.fileproviders.CacheFileProvider
 import expo.modules.imagepicker.fileproviders.CropFileProvider
 import expo.modules.imagepicker.tasks.ImageResultTask
 import expo.modules.imagepicker.tasks.VideoResultTask
-import expo.modules.interfaces.imageloader.ImageLoaderInterface
 import expo.modules.interfaces.permissions.Permissions
 import expo.modules.interfaces.permissions.PermissionsResponse
 import expo.modules.interfaces.permissions.PermissionsResponseListener
 import expo.modules.interfaces.permissions.PermissionsStatus
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.cancel
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 import java.io.IOException
-import java.lang.ref.WeakReference
 
-class ImagePickerModule(
-  private val mContext: Context,
-  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate(),
-  private val pickerResultStore: PickerResultsStore = PickerResultsStore(mContext)
-) : ExportedModule(mContext), ActivityEventListener, LifecycleEventListener {
+private const val moduleName = "ExponentImagePicker"
 
+class ImagePickerModule : Module() {
+  private val context: Context
+    get() = appContext.reactContext.ifNull { throw IllegalStateException("React Application Context is null. You can't access the Context at this point.") }
```

Comment:
I know that we have `ifNull`, but in that case, we can also use `?:`, but  I don't have a strong opinion here.
```suggestion
    get() = appContext.reactContext ?: throw IllegalStateException("React Application Context is null. You can't access the Context at this point.")
```

However, I would change the `IllegalStateException` exception to the custom one (`CodedException`). In the future, we may have logic which will handle it.